### PR TITLE
Mobile app v2.0.1 - User feedback

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
- Addresses User feedback on V2.0.1 of the mobile app

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [x] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Follow instructions in the [README.md](https://github.com/airqo-platform/AirQo-frontend/tree/MOB-55-v-2-0-1-user-feedback/mobile#airqo-mobile-app) to setup your environment. Confirm the done tasks in ticket [MOB-55](https://airqoteam.atlassian.net/browse/MOB-55)

#### What are the relevant tickets?
- [MOB-55](https://airqoteam.atlassian.net/browse/MOB-55)

#### Screenshots (optional)

### CODE REVIEW

**Review by Baalmart — COMMENTED** [review]
Thanks @NoahNsimbe 

1. To easily onboard the rest of the team through PRs, could be awesome if we use review comments in our own PRs to act as code comments that guide the reviewers.
3. Also, it would be awesome if the PR description shared the details of what this PR does.

Thanks\1

File: mobile/lib/auth/email_auth_widget.dart [hunk 1]
```diff
@@ -8,39 +8,43 @@ import 'package:app/widgets/buttons.dart';
...
-    this.appLoading, {
-    Key? key,
-  }) : super(key: key);
+      {Key? key,
+      required this.enableBackButton,
+      required this.changeOption,
+      required this.action,
+      required this.appLoading,
+      required this.emailAddress})
+      : super(key: key);
```
Comment (NoahNsimbe) [inline mobile/lib/auth/email_auth_widget.dart:27]: Refactoring the code to use named parameters

File: mobile/lib/auth/email_auth_widget.dart [hunk 2]
```diff
@@ -8,39 +8,43 @@ import 'package:app/widgets/buttons.dart';
...
   List<String> _emailVerificationCode = <String>['', '', '', '', '', ''];
-  Color _nextBtnColor = Config.appColorDisabled;
-
   final _emailFormKey = GlobalKey<FormState>();
-  final TextEditingController _emailInputController = TextEditingController();
+
+  late TextEditingController _emailInputController;
   late AppService _appService;
+  late String _emailAddress;
+  late Color _nextBtnColor;
```
Comment (NoahNsimbe) [inline mobile/lib/auth/email_auth_widget.dart:47]: organising variables

File: mobile/lib/auth/email_auth_widget.dart [hunk 3]
```diff
@@ -386,8 +395,15 @@ class EmailAuthWidgetState extends State<EmailAuthWidget> {
...
-            'Try logging in');
+            'You already have an '
+            'account with this email address');
+        await Navigator.pushAndRemoveUntil(context,
+            MaterialPageRoute(builder: (context) {
+          return LoginScreen(
+            phoneNumber: '',
+            emailAddress: _emailAddress,
+          );
+        }), (r) => false);
```
Comment (NoahNsimbe) [inline mobile/lib/auth/email_auth_widget.dart:406]: Redirecting a user to the login page together with the email address

File: mobile/lib/auth/email_auth_widget.dart [hunk 4]
```diff
@@ -336,16 +340,21 @@ class EmailAuthWidgetState extends State<EmailAuthWidget> {
 
   void initialize() {
     setState(() {
+      _emailAddress = widget.emailAddress;
+      _nextBtnColor = widget.emailAddress == ''
+          ? Config.appColorDisabled
+          : Config.appColorBlue;
+
```
Comment (NoahNsimbe) [inline mobile/lib/auth/email_auth_widget.dart:347]: Setting default colors for the `Next` button on the sign/login screen

File: mobile/lib/auth/login_screen.dart [hunk 5]
```diff
@@ -24,8 +29,18 @@ class LoginScreenState extends State<LoginScreen> {
...
+                    enableBackButton: false,
+                    changeOption: changeOption,
+                    action: 'login',
+                    appLoading: showLoading,
+                    phoneNumber: widget.phoneNumber)
+                : EmailAuthWidget(
+                    enableBackButton: false,
+                    changeOption: changeOption,
+                    action: 'login',
+                    appLoading: showLoading,
```
Comment (NoahNsimbe) [inline mobile/lib/auth/login_screen.dart:42]: refactoring the auth widgets to use named parameters

File: mobile/lib/auth/phone_auth_widget.dart [hunk 6]
```diff
@@ -477,8 +466,15 @@ class PhoneAuthWidgetState extends State<PhoneAuthWidget> {
...
-              'Try logging in');
+              'You already have an '
+              'account with ths phone number');
+          await Navigator.pushAndRemoveUntil(context,
+              MaterialPageRoute(builder: (context) {
+            return LoginScreen(
+              phoneNumber: '$_countryCode.$_phoneNumber',
+              emailAddress: '',
+            );
+          }), (r) => false);
```
Comment (NoahNsimbe) [inline mobile/lib/auth/phone_auth_widget.dart:477]: Redirecting a user to the login page together with the phone number

File: mobile/lib/services/native_api.dart [hunk 7]
```diff
@@ -240,6 +181,82 @@ class LocationService {
...
+        if (nearestMeasurements.isNotEmpty) {
+          nearestMeasurement = nearestMeasurements.first;
+          for (var measurement in nearestMeasurements) {
+            if (nearestMeasurement!.site.distance > measurement.site.distance) {
+              nearestMeasurement = measurement;
+            }
+          }
+
+          measurements.add(nearestMeasurement!);
+        }
```
Comment (NoahNsimbe) [inline mobile/lib/services/native_api.dart:249]: Getting the second nearest location

**Review by NoahNsimbe — COMMENTED** [review]
cc @Baalmart

**Review by 123MwanjeMike — APPROVED** [review]

File: mobile/lib/services/native_api.dart [hunk 8]
```diff
@@ -240,6 +181,82 @@ class LocationService {
     return await Geolocator.getCurrentPosition();
   }
 
+  // Future<List<Address>> getLocalAddress(double lat, double lang) async {
```
Comment (Baalmart) [inline mobile/lib/services/native_api.dart:184]: we could try to minimise comments in code, unless we are just temporary disabling some functionality or something along those lines.

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**